pycheribuild/projects/release.py

```diff
@@ -0,0 +1,162 @@
+from .cherisim import BuildBeriCtl, BuildCheriSim
+from .project import CheriConfig, SimpleProject
+from ..config.compilation_targets import CompilationTargets
+
+
+repos = ["cheribuild", "cheribsd", "gdb",
+         "morello-llvm-project", "morello-qemu",
+         "morello-trusted-firmware-a", "qemu"]
+
+class Tag(SimpleProject):
+    target = "tag"
+    
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+
+        cls.gittag = cls.add_config_option("gittag", help="Tag to apply")
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+        for repo in repos:
+            run_command("git", "-C", repo, "tag", self.gittag, cwd=source_root)
+
+class UnTag(SimpleProject):
+    target = "untag" 
+
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+
+        cls.gittag = cls.add_config_option("gittag", help="Tag to apply")
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+        for repo in repos:
+            run_command("git", "-C", repo, "tag", "-d", self.gittag, cwd=source_root)
+
+class Release(SimpleProject):
+    project_name = "release"
+    do_not_add_to_targets = True
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+
+        Path(source_root, "cheribuild/cheribuild.json").write_text("""{
+            "source-root": "../../sources",
+            "build-root": "../../build",
+            "output-root": "../../output",
+            "skip-update": true
+} 
+            """)
+
+
+class MorelloRelease(Release):
+    target = "morello-release"
+    dependencies = ["morello-llvm", "cheribsd-morello-purecap",
+                    "gdb-native",
+                    "gdb-morello-hybrid-for-purecap-rootfs",
+                    "arm-none-eabi-toolchain", "morello-acpica",
+                    "morello-scp-firmware",
+                    "morello-trusted-firmware",
+                    "morello-flash-images",
+                    "disk-image-morello-purecap"]
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+
+    def process(self):
+        super().process()
+
+        root = self.config.output_root.parent
+        output_root = self.config.output_root
+
+        install_script = Path(output_root, "install_and_run_fvp.sh")
+        install_script.write_text("""#!/bin/sh
+        dir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+        exec "${dir}/cheribuild.py" install-morello-fvp run-fvp-morello-purecap "$@""")
+        install_script.chmod(0o755)
+
+        run_command("bsdtar", "-cavf", output_root / "release-morello.tar.xz", 
```

Comment:
Should we be putting -purecap in the names? For Morello it's perhaps unnecessary as foo-morello doesn't exist, only foo-morello-{hybrid,purecap}, so one can infer it's purecap, but for RISC-V we do have foo-riscv64 without any suffix that it might get confused with, and we should probably be consistent.